query-engine/connectors/mongodb-query-connector/src/root_queries/mod.rs

```diff
@@ -45,16 +47,37 @@ fn pick_singular_id(model: &ModelRef) -> ScalarFieldRef {
         .unwrap()
 }
 
-pub(crate) async fn metrics<'a, F, T, U>(f: F) -> mongodb::error::Result<T>
+// performs both metrics pushing and query logging. Query logging  might be disabled and thus
+// the query_string might not need to be built, that's why rather than a query_string
+// we receive a Future, as it's not trivial to buid and we want to skip that when possible.
+//
+// As a reminder, the query string is not fed into mongo db directly, we built it for debugging
+// purposes and it's only used when the query log is enabled. For querying mongo, we use the driver
+// wire protocol to build queries from a graphql query rather than executing raw mongodb statements.
+// As we don't have a mongodb query string, we need to create it from the driver object model, which
+// so better skip it if we don't need it, i.e. when the query log is disabled.
+pub(crate) async fn observing<'a, 'b, F, T, U>(
+    query_string_builder: Option<&'b dyn QueryString>,
+    f: F,
+) -> mongodb::error::Result<T>
 where
     F: FnOnce() -> U + 'a,
     U: Future<Output = mongodb::error::Result<T>>,
 {
     let start = Instant::now();
     let res = f().await;
+    let elapsed = start.elapsed().as_millis() as f64;
 
-    histogram!(PRISMA_DATASOURCE_QUERIES_DURATION_HISTOGRAM_MS, start.elapsed());
+    histogram!(PRISMA_DATASOURCE_QUERIES_DURATION_HISTOGRAM_MS, elapsed);
     increment_counter!(PRISMA_DATASOURCE_QUERIES_TOTAL);
 
+    let params: Vec<i32> = Vec::new();
+
+    // todo: emit tracing event with the appropriate log level only query_log is enabled. And fix log suscription
```

Comment:
This is what's out of the scope for the moment. If we want to, I can address this in a following PR.